File: test-plugins/org.yakindu.sct.generator.java.test/src/org/yakindu/sct/generator/java/test/OperationsWithoutBracesCustom.java
/**
 * Copyright (c) 2015 committers of YAKINDU and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     committers of YAKINDU - initial API and implementation
 */

package org.yakindu.sct.generator.java.test;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.*;

import org.yakindu.scr.operationswithoutbraces.OperationsWithoutBracesStatemachine;
import org.yakindu.scr.operationswithoutbraces.IOperationsWithoutBracesStatemachine.SCInterfaceOperationCallback;
import org.yakindu.scr.operationswithoutbraces.OperationsWithoutBracesStatemachine.State;
/**
 *  Unit TestCase for OperationsWithoutBraces
 */
@SuppressWarnings("all")
public class OperationsWithoutBracesCustom implements SCInterfaceOperationCallback{

	private OperationsWithoutBracesStatemachine statemachine;

	@Before
	public void setUp() {
		statemachine = new OperationsWithoutBracesStatemachine();
		statemachine.init();
		statemachine.getSCInterface().setSCInterfaceOperationCallback(this);
	}

	@After
	public void tearDown() {
		statemachine = null;
	}

	@Test
	public void testOperationsWithoutBracesTest() {
		statemachine.enter();
		assertTrue(statemachine.isStateActive(State.main_region_A));
		assertTrue(statemachine.isStateActive(State.another_region_A));
		statemachine.runCycle();
		assertTrue(statemachine.isStateActive(State.main_region_B));
		assertTrue(statemachine.isStateActive(State.another_region_B));
		statemachine.runCycle();
		assertTrue(statemachine.isStateActive(State.main_region_C));
		assertTrue(statemachine.isStateActive(State.another_region_C));
		statemachine.runCycle();
		assertTrue(statemachine.isStateActive(State.main_region_D));
		assertTrue(statemachine.isStateActive(State.another_region_D));
	}
	
	// Default Implementation for Operations
	public boolean myBoolOp() {
		return false;
	}
	public long myIntOp() {
		return 0;
	}
	public double myRealOp() {
		return 0;
	}
}

File: SCT2/trunk/test-plugins/org.yakindu.sct.generator.c.test/gtests/OperationsWithoutBraces/OperationsWithoutBracesCustom.cc
/**
* Copyright (c) 2015 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include <string>
#include "gtest/gtest.h"
#include "OperationsWithoutBraces.h"
#include "OperationsWithoutBracesRequired.h"

OperationsWithoutBraces handle;

TEST(StatemachineTest, OperationsWithoutBracesTest) {
	operationsWithoutBraces_init(&handle);
	operationsWithoutBraces_enter(&handle);
	EXPECT_TRUE(operationsWithoutBraces_isActive(&handle, OperationsWithoutBraces_main_region_A));
	EXPECT_TRUE(operationsWithoutBraces_isActive(&handle, OperationsWithoutBraces_another_region_A));
	operationsWithoutBraces_runCycle(&handle);
	EXPECT_TRUE(operationsWithoutBraces_isActive(&handle, OperationsWithoutBraces_main_region_B));
	EXPECT_TRUE(operationsWithoutBraces_isActive(&handle, OperationsWithoutBraces_another_region_B));
	operationsWithoutBraces_runCycle(&handle);
	EXPECT_TRUE(operationsWithoutBraces_isActive(&handle, OperationsWithoutBraces_main_region_C));
	EXPECT_TRUE(operationsWithoutBraces_isActive(&handle, OperationsWithoutBraces_another_region_C));
	operationsWithoutBraces_runCycle(&handle);
	EXPECT_TRUE(operationsWithoutBraces_isActive(&handle, OperationsWithoutBraces_main_region_D));
	EXPECT_TRUE(operationsWithoutBraces_isActive(&handle, OperationsWithoutBraces_another_region_D));
}
		
sc_boolean operationsWithoutBracesIface_myBoolOp(){
	return false;
}

sc_integer operationsWithoutBracesIface_myIntOp(){
	return 0;
}

sc_real operationsWithoutBracesIface_myRealOp(){
	return 0.0;
}